jcdb-core/src/main/kotlin/org/utbot/jcdb/impl/types/signature/JvmTypes.kt

```diff
@@ -16,54 +16,103 @@
 
 package org.utbot.jcdb.impl.types.signature
 
+import kotlinx.metadata.KmType
 import org.utbot.jcdb.api.PredefinedPrimitives
+import org.utbot.jcdb.api.ext.isNullable
 
-sealed class JvmType {
+sealed class JvmType(val isNullable: Boolean) {
 
     abstract val displayName: String
 
+    abstract fun setNullability(nullable: Boolean): JvmType
```

Comment:
This method looks like setter but should not be setter. Lets stay with `immutable` pattern naming and do something like for class level api: `notNullable()` or something similar